soh/soh/OTRGlobals.cpp

```diff
@@ -530,20 +539,247 @@ extern "C" CollisionHeader* ResourceMgr_LoadColByName(const char* path)
     return (CollisionHeader*)colHeader;
 }
 
-extern "C" Vtx * ResourceMgr_LoadVtxByName(const char* path)
+extern "C" Vtx* ResourceMgr_LoadVtxByName(const char* path)
 {
 	auto res = std::static_pointer_cast<Ship::Array>(OTRGlobals::Instance->context->GetResourceManager()->LoadResource(path));
 	return (Vtx*)res->vertices.data();
 }
 
+extern "C" char* ResourceMgr_LoadSeqByID(int seqID)
+{
+    std::string fmtStr = "audio/sequences/seq_%02X";
+    return OTRGlobals::Instance->context->GetResourceManager()->LoadFile(StringHelper::Sprintf(fmtStr.c_str(), seqID)).get()->buffer.get();
+}
+
+extern "C" int ResourceMgr_GetSeqSizeByID(int seqID)
+{
+    return OTRGlobals::Instance->context->GetResourceManager()
+        ->LoadFile(StringHelper::Sprintf("audio/sequences/seq_%02X", seqID))
+        .get()
+        ->dwBufferSize;
+}
+
+std::map<std::string, SoundFontSample*> cachedCustomSFs;
+
+extern "C" SoundFontSample* ResourceMgr_LoadAudioSample(int romOffset)
+{
+    auto str = StringHelper::Sprintf("audio/samples/sample_%08X", romOffset);
+
+    if (cachedCustomSFs.find(str) != cachedCustomSFs.end())
+        return cachedCustomSFs[str];
+
+    if (romOffset == 0x14f0) {
+        int bp = 0;
+    }
+
+    // Check if our file is actually a wav...
+    auto sampleRaw = OTRGlobals::Instance->context->GetResourceManager()->LoadFile(str);
+    uint32_t* strem = (uint32_t*)sampleRaw->buffer.get();
+    uint8_t* strem2 = (uint8_t*)strem;
+
+    if (strem2[0] == 'R' && strem2[1] == 'I' && strem2[2] == 'F' && strem2[3] == 'F')
+    {
+        SoundFontSample* sampleC = (SoundFontSample*)malloc(sizeof(SoundFontSample));
```

Comment:
We can use `new SoundFontSample` instead of malloc since we use C++. That will either throw an error or terminate rather than returning null (which is currently not handled).